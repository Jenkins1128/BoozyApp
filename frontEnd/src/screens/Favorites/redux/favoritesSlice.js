import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
