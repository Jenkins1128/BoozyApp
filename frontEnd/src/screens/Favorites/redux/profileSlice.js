import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
