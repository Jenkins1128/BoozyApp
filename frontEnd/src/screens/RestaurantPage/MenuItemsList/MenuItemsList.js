import React from 'react';
import { View, Text, FlatList, TouchableOpacity, StyleSheet } from 'react-native';
import MenuItem from '../MenuItem/MenuItem';

const MenuItemsList = ({ state, showMenuItemOverlay, dismiss }) => {
	const menuItem = ({ item, index }) => {
		return <MenuItem key={index} keyval={index} val={item} dismiss={dismiss} />;
	};
	return (
		<View testID={'scrollContainer'} style={styles.scrollContainer}>
			<View testID={'menuItemsTitleContainer'} style={styles.menuItemsTitleContainer}>
				<Text testID={'menuTitle'} style={styles.menuTitle}>
					Menu items
				</Text>
				<View testID={'menuButtonContainer'} style={styles.menuButtonContainer}>
					<TouchableOpacity testID={'menuButton'} onPress={() => showMenuItemOverlay(true)} style={styles.menuButton}>
						<Text testID={'menuItemText'} style={styles.menuItemText}>
							+ Menu Item
						</Text>
					</TouchableOpacity>
				</View>
			</View>

			<FlatList data={state.menuItemArray} keyExtractor={(item, i) => i.toString()} renderItem={menuItem}></FlatList>
		</View>
	);
};

export default MenuItemsList;

const styles = StyleSheet.create({
	scrollContainer: {
		flex: 1,
		width: '100%',
		backgroundColor: 'white'
	},
	menuItemsTitleContainer: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center'
	},
	menuTitle: {
		color: 'lightgrey',
		fontSize: 20,
		marginTop: 0,
		fontWeight: 'bold',
		padding: 10
	},
	menuButtonContainer: {
		marginRight: 15
	},
	menuButton: {
		width: 110,
		backgroundColor: '#EB8873',
		borderRadius: 20,
		padding: 7
	},
	menuItemText: {
		color: 'white',
		fontWeight: 'bold'
	}
});
