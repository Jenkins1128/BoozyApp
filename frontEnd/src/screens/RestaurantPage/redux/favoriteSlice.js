import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const favoriteAsync = createAsyncThunk('favoriteAsync/status', async (data, { rejectWithValue }) => {
	const { restaurantId, name } = data;
	try {
		//favorite restaurant
		await axios.post(`${baseUrl}/user/add`, {
			restaurantId: restaurantId,
			name: name
		});
		//add restaurants to your favorites
		const response = await axios.post(`${baseUrl}/favorites`, {
			restaurantId: restaurantId,
			name: name
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const favoriteSlice = createSlice({
	name: 'favoriteAsync',
	initialState,
	reducers: {
		resetFavorite: (state) => {
			state.currentState.favoriteRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(favoriteAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					favoriteColor: payload[0]['contains'] ? 'red' : 'white',
					favoriteRequestStatus: 'fulfilled'
				});
			})
			.addCase(favoriteAsync.rejected, (state) => {
				state.currentState.favoriteRequestStatus = 'rejected';
			});
	}
});

export const { resetFavorite } = favoriteSlice.actions;

export const selectFavoriteState = (state) => state.favorite.currentState;
export default favoriteSlice.reducer;
