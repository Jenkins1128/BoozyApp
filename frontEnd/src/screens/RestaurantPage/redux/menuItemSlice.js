import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const postMenuItemAsync = createAsyncThunk('postMenuItemAsync/status', async (data, { rejectWithValue }) => {
	const { description, price, restaurantName, restaurantId } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/menu', {
			content: description,
			price: price,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const postMenuItemSlice = createSlice({
	name: 'postMenuItemAsync',
	initialState,
	reducers: {
		updatePrice: (state, { payload }) => {
			state.currentState = { ...state.currentState, price: payload.price };
		},
		updateDescription: (state, { payload }) => {
			state.currentState = { ...state.currentState, description: payload.description };
		},
		resetMenuItem: (state) => {
			state.currentState = { ...state.currentState, price: '', description: '', postMenuItemRequestStatus: 'idle' };
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(postMenuItemAsync.fulfilled, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'fulfilled' };
			})
			.addCase(postMenuItemAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'rejected' };
			});
	}
});

export const { updatePrice, updateDescription, resetMenuItem } = postMenuItemSlice.actions;

export const selectMenuItemState = (state) => state.menuItem.currentState;
export default postMenuItemSlice.reducer;
