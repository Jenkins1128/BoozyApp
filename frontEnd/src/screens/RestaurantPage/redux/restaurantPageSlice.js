import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		restaurantId: '',
		restaurantName: '',
		restaurantImage: 'https://boozyimage.s3.us-east-2.amazonaws.com/logo.png',
		phoneNumber: '',
		allCategories: '',
		rating: 0.0,
		reviews: 0,
		favoriteColor: 'white',
		changeFavoritedColor: '',
		starCount: 0,
		favorited: false,
		rated: false,
		price: 0.0,
		description: '',
		menuItemArray: []
	}
};

export const getMenuItemsAsync = createAsyncThunk('getMenuItemsAsync/status', async (restaurantId, { rejectWithValue }) => {
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/menu/list', {
			restaurantId: restaurantId
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const restaurantPageSlice = createSlice({
	name: 'getMenuItemsAsync',
	initialState,
	reducers: {
		updateState: (state, { payload }) => {
			console.log('payload', payload);
			state.currentState = {
				...state.currentState,
				restaurantId: payload.params.id,
				restaurantName: payload.params.name,
				restaurantImage: payload.params.img,
				phoneNumber: payload.params.phone,
				allCategories: payload.params.allCategories,
				rating: payload.params.rating,
				reviews: payload.params.reviews,
				rated: payload.params.alreadyRated,
				favorited: payload.params.alreadyFavorited,
				starCount: 0
			};
		},
		updateFavoriteColor: (state, { payload }) => {
			state.currentState = {
				...state.currentState,
				favoriteColor: payload.color
			};
		},
		updateStarCount: (state, { payload }) => {
			state.currentState = {
				...state.currentState,
				starCount: payload.starCount
			};
		},
		updateMenuItems: (state, { payload }) => {
			state.currentState = { ...state.currentState, menuItemArray: [...state.currentState.menuItemArray, payload.menuItem] };
		},
		resetRestaurantPageRequestStatus: (state) => {
			state.currentState.restaurantPageRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getMenuItemsAsync.fulfilled, (state, { payload }) => {
				state.currentState = { ...state.currentState, restaurantPageRequestStatus: 'fulfilled', menuItemArray: payload };
			})
			.addCase(getMenuItemsAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, restaurantPageRequestStatus: 'rejected' };
			});
	}
});

export const { updateState, updateFavoriteColor, updateStarCount, updateMenuItems, resetRestaurantPageRequestStatus } = restaurantPageSlice.actions;

export const selectRestaurantPageState = (state) => state.restaurantPage.currentState;
export default restaurantPageSlice.reducer;
