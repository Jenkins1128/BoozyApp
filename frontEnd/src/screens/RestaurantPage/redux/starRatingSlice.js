import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const starRatingAsync = createAsyncThunk('starRatingAsync/status', async (data, { rejectWithValue }) => {
	const { rating, restaurantId, restaurantName } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/restaurants', {
			content: rating,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const starRatingSlice = createSlice({
	name: 'starRatingAsync',
	initialState,
	reducers: {
		resetStarRatingRequestStatus: (state) => {
			state.currentState.starRatingRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(starRatingAsync.fulfilled, (state) => {
				state.currentState.starRatingRequestStatus = 'fulfilled';
			})
			.addCase(starRatingAsync.rejected, (state) => {
				state.currentState.starRatingRequestStatus = 'rejected';
			});
	}
});

export const { resetStarRatingRequestStatus } = starRatingSlice.actions;

export const selectStarRatingState = (state) => state.starRating.currentState;
export default starRatingSlice.reducer;
