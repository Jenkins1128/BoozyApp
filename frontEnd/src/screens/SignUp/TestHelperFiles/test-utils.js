import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
