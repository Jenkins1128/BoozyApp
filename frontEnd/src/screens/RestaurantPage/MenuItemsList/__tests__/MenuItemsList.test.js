import React from 'react';
import { cleanup, render, fireEvent } from '../../../../TestHelperFiles/redux/test-utils';
import MenuItemsList from '../MenuItemsList';

describe('<MenuItemsList />', () => {
	let state;
	let showMenuItemOverlay = jest.fn(),
		dismiss = jest.fn();
	let rendered;
	beforeEach(() => {
		state = {
			menuItemArray: []
		};
		(showMenuItemOverlay = jest.fn()), (dismiss = jest.fn());
		rendered = render(<MenuItemsList state={state} showMenuItemOverlay={showMenuItemOverlay} dismiss={dismiss} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap flexible wrapper over view and list', () => {
		const scrollContainerViewComponent = rendered.getByTestId('scrollContainer');
		const given = scrollContainerViewComponent.props.style;
		const result = {
			flex: 1,
			width: '100%',
			backgroundColor: 'white'
		};
		expect(result).toMatchObject(given);
	});

	it('should render menuItemsTitleContainer styles correctly', () => {
		const menuItemsTitleContainerComponent = rendered.getByTestId('menuItemsTitleContainer');
		const given = menuItemsTitleContainerComponent.props.style;
		const result = {
			flexDirection: 'row',
			justifyContent: 'space-between',
			alignItems: 'center'
		};
		expect(result).toMatchObject(given);
	});

	it('should render menuTitle styles correctly', () => {
		const menuTitleComponent = rendered.getByTestId('menuTitle');
		const given = menuTitleComponent.props.style;
		const result = {
			color: 'lightgrey',
			fontSize: 20,
			marginTop: 0,
			fontWeight: 'bold',
			padding: 10
		};
		expect(result).toMatchObject(given);
	});

	it('should render menuButtonContainer styles correctly', () => {
		const menuButtonContainerComponent = rendered.getByTestId('menuButtonContainer');
		const given = menuButtonContainerComponent.props.style;
		const result = {
			marginRight: 15
		};
		expect(result).toMatchObject(given);
	});

	it('should render menu button styles correctly', () => {
		const menuButtonComponent = rendered.getByTestId('menuButton');
		const given = menuButtonComponent.props.style;
		const result = {
			width: 110,
			backgroundColor: '#EB8873',
			borderRadius: 20,
			padding: 7,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render menuItemText styles correctly', () => {
		const menuItemTextComponent = rendered.getByTestId('menuItemText');
		const given = menuItemTextComponent.props.style;
		const result = {
			color: 'white',
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	//interaction
	it('should fire menu button event', () => {
		const menuButton = rendered.getByTestId('menuButton');
		fireEvent.press(menuButton);
		expect(showMenuItemOverlay).toHaveBeenCalledTimes(1);
	});
});
