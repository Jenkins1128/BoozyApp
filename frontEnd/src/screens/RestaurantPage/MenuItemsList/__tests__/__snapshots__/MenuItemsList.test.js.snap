// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItemsList /> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "width": "100%",
    }
  }
  testID="scrollContainer"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    testID="menuItemsTitleContainer"
  >
    <Text
      style={
        Object {
          "color": "lightgrey",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 0,
          "padding": 10,
        }
      }
      testID="menuTitle"
    >
      Menu items
    </Text>
    <View
      style={
        Object {
          "marginRight": 15,
        }
      }
      testID="menuButtonContainer"
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#EB8873",
            "borderRadius": 20,
            "opacity": 1,
            "padding": 7,
            "width": 110,
          }
        }
        testID="menuButton"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
          testID="menuItemText"
        >
          + Menu Item
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
